### ISSUE TITLE
Is it possible to query which Xinput device each DS4 controller was assigned?

### ISSUE THREAD

**Issue by bernspedras** [Ryochan7/DS4Windows#2069]
TL;DR version -> I would like to query (via DS4WindowsCMD) something like this 
> DS4WindowsCmd.exe -command query.1.xinputindex   (and returns the xinputindex of the virtual Xinput device created for slot1)

or (if easier)
> DS4Windows.Cmd.exe -command query.1.xinputHID (returns the HID DevicePath of the virtual XinputDevice)

For Context:

I'm developing a solution to auto configure many emulators with any combination of controllers..(DirectInput/Xinput/ DS4 etc..) I use DS4Windows for 2 reasons 
1 - To use XInput with DS4 controllers (for analog triggers and easier setup on some emulators)
2 - To use DSUClient for motion input for Wii and WiiU emulators

My situation: I have 2 (real) Xbox 360 controller connected.. one is Xinput index 1 and the other is Xinput index 2. When a launch DS4Windows, and connect my DS4, it creates a virtual controller and is recognized as Xinput index 3. (this "3" is the index I'm looking for)

The indexes may vary depending on the order I connect them. 

On the GUI of DS4Windows there is no indecation of other connected XInputDevices or of the corresponding index...

is there a way to find this information?

**Comment by bernspedras** [comment]
poking around to find a solution, I ended up in the file Xbox360OutDevice.cs.. in the constructor I see:

        public Xbox360OutDevice(ViGEmClient client)
        {
            cont = client.CreateXbox360Controller();
            var myIndex = cont.UserIndex; //Added by me --> throws excpetion!
            cont.AutoSubmitReport = false;
        }
I think we depend on vigemclient to inform us the index... it even seems to have the property... but it throws:

Exception of type 'Nefarius.ViGEm.Client.Targets.Xbox360.Exceptions.Xbox360UserIndexNotReportedException' was thrown.

does it help? is it possible to get the index?

**Comment by bernspedras** [comment]
Got it working! After poking around ViGEm code, I noticed that the UserIndex property is filled "late" (after a notification event).
So, my guess is that immediately after the constructor call, the property is still not set. Im not sure when it is set.. but if calles later (say from a command from DS4WindowsCmd) it does work!

I'll prepare a pull request with a new cmd line query like this:
DS4WindowsCmd.exe -command query.1.outputXinputIndex

I hope it also gets merge since my solution depends on it :)

**Comment by Ryochan7** [comment]
In short: No.

That property outright does not work most of the time in ViGEmBus 1.16.112.0. It is only reliable to use it when using ViGEmBus 1.17.333.0. At this rate, I figure I am going to have to, at least temporarily, reduce the restriction on ViGEmBus to allow version 1.16.112.0 to be usable.

**Comment by Ryochan7** [comment]
Already not so sure. #2067 shows that I cannot rely on people to give me the right info.
